xray/handler.go

```diff
@@ -125,7 +127,7 @@ func httpTrace(seg *Segment, h http.Handler, w http.ResponseWriter, r *http.Requ
 
 	cLen, _ := strconv.Atoi(capturer.Header().Get("Content-Length"))
 	seg.GetHTTP().GetResponse().ContentLength = cLen
```

Comment:
Probably better if we can reduce this to one line like below. Also, I think we should add locks here.

```
seg.Lock()
seg.GetHTTP().GetResponse().ContentLength = strconv.Atoi(capturer.Header().Get("Content-Length"))
seg.Unlock()
```